perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ExpirationNotifScheduler.java

```diff
@@ -357,6 +378,82 @@ private void checkVoMembersState(List<Vo> vos) throws WrongAttributeAssignmentEx
 		expireSponsorships();
 	}
 
+	/**
+	 * Checks all applications for given vos and if some application is expired (according to expiration rules set by
+	 * VO Manager), then rejects this application.
+	 *
+	 * @param vos virtual organizations
+	 * @throws PerunException
+	 */
+	private void voApplicationsAutoRejection(List<Vo> vos) throws PerunException {
```

Comment:
Generally it would be better to try/catch exceptions on each application rejection rather than the bulk of them or all vos. Like this, one exception will discard processing of all other applications, vos and groups.

Reply:
The rejection process is in method `rejectExpiredApplications` so I added try/catch block here. Is it what you meant?

Reply:
Yes, thank you.